### PR TITLE
Add / New error message added for user rejected transaction on wallet

### LINKED ISSUE #98: Add improvement to formatError
Let's bring the improvement on user rejected error to the `formatError` function.

Implementation:
https://github.com/liteflow-labs/o-mee/pull/435/files/603f9ef16dc4443df922482106241038795ce2e0#diff-b01fbf78bcecc7bcfbabc4d282c48ed3ee1227e9e13af8a1609bdfc04f135a55

_Originally posted by @NicolasMahe in https://github.com/liteflow-labs/o-mee/pull/435#discussion_r1051770697_

### PR DESCRIPTION
**Pull request by ismailToyran**
Trello card: https://trello.com/c/jw4bXigZ/452-bring-error-improvement-to-nft-hooks
Closes #98 
After this merge, could change O-mee and remove extra function which has been implemented on https://github.com/liteflow-labs/o-mee/pull/435#discussion_r1051770697

### CODE REVIEW

File: packages/hooks/src/errorMessages.ts [hunk 1]
```diff
@@ -34,4 +34,6 @@ export enum ErrorMessages {
   OWNERSHIP_NOT_FOUND = 'Ownership not found',
 
   POLLING_TIMEOUT = 'Polling timeout. could not check ownership',
+
+  TRANSACTION_REJECTED_BY_USER = 'You rejected the transaction',
```
Comment (antho1404) [inline packages/hooks/src/errorMessages.ts]: All error messages are neutral without saying "You"

**Review by antho1404 — COMMENTED** [review]
I'm not sure this makes sense the formatting of the error is just to format the error not transform and change the behavior of the error. 
Right now if an application wants to get the original message, they can't. The error should be as "pure" as possible as long as possible and only at the last moment to display to the user it should be transformed if needed which can even nicely be done with translation based on the code of the error.
I would really suggest to keep it as it is for now

**Comment by NicolasMahe** [comment]
> I'm not sure this makes sense the formatting of the error is just to format the error not transform and change the behavior of the error.

I'm not sure to follow, the function already transform an `Error` object to a `string`.

> Right now if an application wants to get the original message, they can't. 

That's basically what the function is already doing. The recursive call to extract the deepest error's message already loose the original error which is the top one.

---

The big question I have with `formatError` is: does it make sense to have it in the hooks package?
The current use of this function is to create an error message that can be display to the user. This is UI stuff and should be in starter-kit.

EDIT:
Looking at https://github.com/liteflow-labs/liteflow-js/issues/26, `formatError` is in the category "To move out of hooks package or to remove if not used"

**Comment by antho1404** [comment]
Agree. That kind of goes in the same direction where this should be as close to the UI as possible.
Also, I believe most of the utils are not actually used in the hook, so could/should be moved and then in that case, this seems unnecessary, and the current way to have that in the app is the proper way like it's done already in the related issue

**Comment by NicolasMahe** [comment]
I'm good to merge this PR and quickly move this and other helper functions directly in the starter-kit. It will be easier to track.

File: packages/hooks/src/errorMessages.ts (hunk 1 above)
Comment (NicolasMahe) [reply packages/hooks/src/errorMessages.ts]:
```suggestion
  TRANSACTION_REJECTED_BY_USER = 'Transaction was cancelled',
```

**Review by NicolasMahe — APPROVED** [review]

**Pull request merged**